src/storage/Bucket.ts

```diff
@@ -108,15 +121,16 @@ export class Bucket {
         this._updateTTL()
     }
 
-    _updateTTL() {
+    private _updateTTL(): void {
         this.ttl = new Date()
         this.ttl.setSeconds(this.ttl.getSeconds() + this._keepAliveSeconds)
         this.logger.debug(`new ttl: ${this.ttl}`)
     }
 
-    isAlive() {
-        const isAlive = this.ttl >= new Date()
-        this.logger.debug(`isAlive: ${isAlive}, ${this.ttl} >= ${new Date()}`)
-        return this.ttl >= new Date()
+    isAlive(): boolean {
+        const now = new Date()
+        const isAlive = this.ttl >= now
+        this.logger.debug(`isAlive: ${isAlive}, ${this.ttl} >= ${now}`)
+        return this.ttl >= now
```

Comment:
This log was potentially misleading since it was generating 3 different `new Date()` values, which could potentially have 3 different dates depending on how slow the CPU was running and how close they were to the next millisecond.

e.g. it could log isAlive: false, 100 > 99 but then actually return `true` in  `this.ttl >= new Date()`

*Never assume successive calls to new Date or Date.now() will produce the same value.*

Reply:
Yup 100%